Source/DiffableCollection/FetchedCollectionDiffableDataSource.swift

```diff
@@ -0,0 +1,206 @@
+	/// A boolean indicating whether snapshot updates should be applied immediately. Changes occuring to the snapshot will remain pending until the next time this value is set to `true`. The default value is `true`.
+	public var isUpdatingAutomatically: Bool {
+		didSet {
+			guard isUpdatingAutomatically else { return }
+			applyPendingSnapshot()
+		}
+	}
+
+	private let controller: NSFetchedResultsController<NSFetchRequestResult>
+	private let dataSource: FetchedDiffableDataSource
+	private lazy var internalSnapshot: NSDiffableDataSourceSnapshot<FetchedDiffableSection, FetchedDiffableItem> = .init() // non-modifiable snapshot to maintain data integrity, this snapshot stays in sync with the database
+	private var pendingSnapshot: NSDiffableDataSourceSnapshot<FetchedDiffableSection, FetchedDiffableItem>? // store pending changes while `isUpdatingAutomatically` is set to `false`
+	private weak var delegate: FetchedCollectionDiffableDataSourceDelegate?
+
+	// MARK: - Lifecycle
+
+	/// Creates a diffable data source based on a `NSFetchedResultsController` instance.
+	public init(controller: NSFetchedResultsController<NSFetchRequestResult>, dataSource: FetchedDiffableDataSource, delegate: FetchedCollectionDiffableDataSourceDelegate? = nil, isHidingEmptySections: Bool = true, isUpdatingAutomatically: Bool = true) {
+		self.controller = controller
+		self.dataSource = dataSource
+		self.delegate = delegate
+		self.isHidingEmptySections = isHidingEmptySections
+		self.isUpdatingAutomatically = isUpdatingAutomatically
+		super.init()
+		commonInit()
+	}
+
+	private func commonInit() {
+		setDelegate()
+		performFetch()
+	}
+
+	// MARK: - Controller management
+
+	/// Sets the delegate of the NSFetchedResultsController instance.
+	private func setDelegate() {
+		controller.delegate = self
+	}
+
+	/// Executes the fetch request of the NSFetchedResultsController instance.
+	private func performFetch() {
+		do {
+			try controller.performFetch()
+		} catch let error {
+			assertionFailure("Error performing controller fetch: \(error)")
+		}
+	}
+
+	// MARK: - NSFetchedResultsControllerDelegate
+
+	public func controller(_ controller: NSFetchedResultsController<NSFetchRequestResult>, didChangeContentWith snapshot: NSDiffableDataSourceSnapshotReference) {
+		var itemsBeforeChange = internalSnapshot.itemIdentifiers
+		let typedSnapshot = transform(snapshot: snapshot, context: controller.managedObjectContext)
+		updateInternalSnapshot(withChangedContent: typedSnapshot, maintainSectionOrder: !(controller.sectionNameKeyPath?.isEmpty ?? true)) // maintain section order for FRC's with `sectionNameKeyPath` set
+		var externalSnapshot = internalSnapshot // modifiable snapshot to be displayed
+		delegate?.contentChangeWillBeApplied(snapshot: &externalSnapshot)
+
+		var itemsAfterChange = externalSnapshot.itemIdentifiers
+		var itemsToReconfigure = Array(Set(itemsBeforeChange).intersection(Set(itemsAfterChange)))
+		if #available(iOS 15.0, *), isReconfiguringItems {
```

Comment:
I don't think we should call `reconfigureItems` or `reloadItems` manually if we transfer transfer these properties from the original snapshot, given that is something that the FRC provides.